lib/model/EcoscoreData.dart

```diff
@@ -27,4 +27,7 @@ class EcoscoreData extends JsonObject {
 
   @override
   Map<String, dynamic> toJson() => _$EcoscoreDataToJson(this);
+
+  static Map<String, dynamic> toJsonHelper(EcoscoreData d) =>
```

Comment:
I see your problem and agree that it does not look normal to have a EcoscoreData instance in a json serialization output.

However, I'm a bit puzzled because: 
1/ the official sample seems to behave the same (line 39 of https://github.com/google/json_serializable.dart/blob/master/example/lib/example.g.dart we can see that it directly put the 'instance.orders' variable, without using toJson on it)

2/ I have an app which has the same "issue", but when I give the object to retrofit it will simply work. At the end, and I'm not sure why, the server receives a correct json.

So I wonder if it has not been designed this way, even if I find that quite surprising as well... Does someone has an answer?

Reply:
Ok, I just found an explanation there: https://github.com/google/json_serializable.dart/issues/702

We need to use 'explicitToJson: true' on all @JsonSerializable classes (including Product) to "fix" this issue.
But in explicitToJson documentation they say it's not really needed to do so since dart:convert will handle it.

So either we do nothing and accept this behavior, either we use explicitToJson everywhere.

Reply:
Hi @FredJul , I am aware of the `explicitToJson` flag I intentionally did not add it. Currently, the SDK simply handles JSON generation differently (using the `toJson` helper function) and I wanted to fix the bug in a way that is consistent with respect to how JSON generation is handled for other classes/fields in the SDK.

Note, I'm not saying the way the SDK handles JSON generation is "right". In fact, using the `explicitToJson` could be a better way going forward. If we agree, at a later point, to use `explicitToJson` then we can add that using a separate pull request. That would require addressing this problem for all JSON classes in the SDK, which is outside the scope of this particular, PR.

For what is worth this PR is consistent with the way JSON generation is handled for other classes and fields in this SDK.

Reply:
Well, this PR seems to be consistent with the rest of the app, but I'm not fan of the added complexity (compared of doing nothing, or using `explicitToJson` if that's necessary).
However, I'm quite a new guy here and I may have missed something. Maybe others reviewers like @PrimaelQuemerais will agree with you.

Reply:
I approved it for now, we still need to write the best practices document for the project, we should probably discuss this at some point. I will take a look at the `explicitToJson` parameter